### PR TITLE
Allow per_try_timeout when global route timeout is disabled.

### PR DESCRIPTION
**Pull request by ggreenway**
This allows, for instance, a per_try_timeout (which is only in
effect until the response to downstream begins) with streaming endpoints,
for which an overall timeout doesn't make sense.

Signed-off-by: Greg Greenway <ggreenway@apple.com>

Risk Level: Low
Testing: New tests written
Docs Changes: clarified docs
Release Notes: added

### CODE REVIEW

**Comment by mattklein123** [comment]
In addition to clarify the option docs themselves, some discussion in https://www.envoyproxy.io/docs/envoy/latest/faq/configuration/timeouts would be great also.

File: docs/root/configuration/http/http_filters/router_filter.rst [hunk 1]
```diff
@@ -240,7 +240,8 @@ x-envoy-upstream-rq-per-try-timeout-ms
 
 Setting this header on egress requests will cause Envoy to set a *per try* timeout on routed
 requests. This timeout must be <= the global route timeout (see
-:ref:`config_http_filters_router_x-envoy-upstream-rq-timeout-ms`) or it is ignored. This allows a
```
Comment (snowp) [inline docs/root/configuration/http/http_filters/router_filter.rst]: Actually, could you reword this to something like `If a global timeout is defined .. this must be <= the global timeout or it will be ignored`? I think it would read a bit cleaner

Comment (ggreenway) [reply docs/root/configuration/http/http_filters/router_filter.rst]: Agreed, that will flow better. Also, I just noticed that `<=` on the next line is incorrect. It should be `<`. If they're equal, it will still be ignored. I'll fix that while I'm touching it.

**Comment by ggreenway** [comment]
> LGTM, but seems like CI is red

Weird. No code changes at all, but it failed to compile. I'll merge master and see if that fixes it.
